### ISSUE TITLE
Camera animation: onCancel() in CancelableCallback never called

### ISSUE THREAD

**Issue by mhettig** [mapbox/mapbox-gl-native#6910]
Hey,

as mentioned in the title I'm a little confused about the behavior of the `onCancel()` method in the `MapboxMap.CancelableCallback()`. It never seems to be called when stopping a camera animation e.g. by user input (note the code snippet below to see my implementation). Currently I'm running on `mapbox-android-sdk:4.2.0-beta.4`.

According to the API documentation `onCancel()` should be called:

> If the animation stops due to interruption by [...] a user gesture [(1)](https://www.mapbox.com/android-sdk/api/4.1.0/com/mapbox/mapboxsdk/maps/MapboxMap.html#animateCamera-com.mapbox.mapboxsdk.camera.CameraUpdate-int-com.mapbox.mapboxsdk.maps.MapboxMap.CancelableCallback-)

or if the animation did not complete normally. [(2)](https://www.mapbox.com/android-sdk/api/4.1.0/com/mapbox/mapboxsdk/maps/MapboxMap.html#animateCamera-com.mapbox.mapboxsdk.camera.CameraUpdate-com.mapbox.mapboxsdk.maps.MapboxMap.CancelableCallback-)

So the **expected behavior** would be: `onCancel()` get's called if interrupting the camera animation by a user gesture.

But the **actual behavior** is quite more strange: Regardless of whether the animation finishes as planned or gets interrupted `onFinished()` gets called making it impossible to distinguish between a successful completed animation or an incomplete one.

_Example code:_
```java
CameraPosition newPosition = new CameraPosition.Builder()
    .target(new LatLng(mMapBoxMap.getMyLocation()))
    .zoom(14)
    .build();

mMapBoxMap.animateCamera(CameraUpdateFactory.newCameraPosition(newPosition), 2000,
    new MapboxMap.CancelableCallback()
    {
        @Override
        public void onCancel()
        {
        ...
        }

        @Override
        public void onFinish()
        {
        ...
        }
});
```

**Comment by tobrun** [comment]
I noticed the same and implemented a [fix](https://github.com/mapbox/mapbox-gl-native/pull/6891) for this yesterday. This was merged with the 4.2.0 release branch and will be available with next snapshot/beta/release. Thank you for reaching out!

**Comment by mhettig** [comment]
Okay cool. Thank you!
